src/lsm/schema.zig

```diff
@@ -458,10 +460,30 @@ pub const Manifest = struct {
         assert(context.entry_count == @divExact(header.size - @sizeOf(vsr.Header), entry_size));
         assert(stdx.zeroed(&context.reserved));
 
+        if (header.commit == 0) {
+            assert(header.parent == 0);
+        }
+
         return .{ .entry_count = context.entry_count };
     }
 
-    pub fn size(schema: *const Manifest) u32 {
+    pub fn previous(manifest_block: BlockPtrConst) ?struct { checksum: u128, address: u64 } {
```

Comment:
We _could_ pass-in the head we know from the superblock, so that we can return `null` it it is really `null` or just dangling. 

The beneifit here is that the call-site will be aware that the chain of blocks might be cut short.

Though, this feels like overengieering to be honest, the current API is simple (but maybe add a comment that this might point to a garbage block?)

Reply:
I am also wondering -- by the time we reach the last block when opening manifest, can we assert anything about that entry with respect to the free set? Ie, that the precious entry should always be free?

Reply:
> I am also wondering -- by the time we reach the last block when opening manifest, can we assert anything about that entry with respect to the free set? Ie, that the precious entry should always be free?

I think that would be possible right now, but it definitely won't work once manifest compaction pacing is in (next PR).
The reason is that when the manifest is small (has few blocks) the manifest might decide to not do any compaction during a particular half-bar. That gives the freeset the opportunity to reuse the block even while the oldest manifest block still references it.